geom/perf_test.go

```diff
@@ -421,3 +422,65 @@ func BenchmarkMultiLineStringIsSimpleManyLineStrings(b *testing.B) {
 		})
 	}
 }
+
+func BenchmarkForceCWandForceCCWOrientedCorrectly(b *testing.B) {
+	for i, tc := range []struct {
+		wkt     string
+		geoType GeometryType
+		isCW    bool
+		isCCW   bool
+		note    string
+	}{
+		{"POLYGON((0 0,0 5,5 5,5 0,0 0))", TypePolygon, true, false, "CW"},
+		{"POLYGON((1 1,3 1,2 2,2 4,1 1))", TypePolygon, false, true, "CCW"},
+		{"POLYGON((0 0,0 5,5 5,5 0,0 0), (1 1,3 1,2 2,2 4,1 1))", TypePolygon, true, false, "outer CW inner CCW"},
+		{"POLYGON((0 0,5 0,5 5,0 5,0 0), (1 1,1 2,2 2,2 1,1 1))", TypePolygon, false, true, "outer CCW inner CW"},
+		{"MULTIPOLYGON(((40 40, 45 30, 20 45, 40 40)),((20 35, 45 20, 30 5, 10 10, 10 30, 20 35),(30 20, 20 25, 20 15, 30 20)))", TypeMultiPolygon, true, false, "all CW"},
+		{"MULTIPOLYGON(((40 40, 20 45, 45 30, 40 40)),((20 35, 10 30, 10 10, 30 5, 45 20, 20 35),(30 20, 20 15, 20 25, 30 20)))", TypeMultiPolygon, false, true, "all CCW"},
+		{"GEOMETRYCOLLECTION(POLYGON((0 0,0 5,5 5,5 0,0 0)), MULTIPOLYGON(((40 40, 45 30, 20 45, 40 40)),((20 35, 45 20, 30 5, 10 10, 10 30, 20 35),(30 20, 20 25, 20 15, 30 20))))", TypeGeometryCollection, true, false, "all CW"},
+	} {
+		b.Run(strconv.Itoa(i), func(b *testing.B) {
```

Comment:
What do you think about collapsing the two tests into one? I noticed that the tables are basically the same. You could do something like this:

```
b.Run(strconv.Itoa(i), func(b *testing.B) {
    g := geomFromWKT(b, tc.wkt)
    for correct, label := range map[bool]string{
        "correct":   true,
        "incorrect": false,
    } {
        b.Run(label, func(b *testing.B) {
            b.ResetTimer()
            for i := 0; i < b.N; i++ {
                if tc.isCW && correct || tc.isCCW && !correct {
                    g.ForceCW()
                } else if tc.isCCW && correct || tc.IsCW && !incorrect {
                    g.ForceCCW()
                }
            }
        })
    }
})
```

Thoughts?

Reply:
Done!